File: frontend/src/types/job.ts
/** Job position types for admin CRUD */

export interface JobListItem {
  id: string
  title: string
  category: string
  level: string
  salary_range: string | null
  company_type: string
  is_active: boolean
}

export interface SkillNode {
  name: string
  level?: 'easy' | 'medium' | 'hard'
  children?: SkillNode[]
}

export interface SkillTree {
  domains: {
    name: string
    weight: number
    skills: SkillNode[]
  }[]
}

export interface JobDetail extends JobListItem {
  description: string | null
  requirements: Record<string, unknown>
  skill_tree: SkillTree | null
  created_at: string
}

export interface JobCreate {
  title: string
  category: string
  level?: string
  description?: string | null
  requirements: Record<string, unknown>
  skill_tree?: SkillTree | null
  salary_range?: string | null
  company_type?: string
  is_active?: boolean
}

export interface JobUpdate {
  title?: string | null
  category?: string | null
  level?: string | null
  description?: string | null
  requirements?: Record<string, unknown> | null
  skill_tree?: SkillTree | null
  salary_range?: string | null
  company_type?: string | null
  is_active?: boolean | null
}

File: backend/app/schemas/job.py
"""Job schemas."""
from datetime import datetime
from pydantic import BaseModel, Field


class JobOut(BaseModel):
    id: str
    title: str
    category: str
    level: str
    description: str | None = None
    requirements: dict
    skill_tree: dict | None = None
    salary_range: str | None = None
    company_type: str
    is_active: bool
    created_at: datetime

    model_config = {"from_attributes": True}


class JobListItem(BaseModel):
    id: str
    title: str
    category: str
    level: str
    salary_range: str | None = None
    company_type: str
    is_active: bool

    model_config = {"from_attributes": True}


class JobCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=100)
    category: str = Field(..., min_length=1, max_length=50)
    level: str = Field(default="mid")
    description: str | None = None
    requirements: dict
    skill_tree: dict | None = None
    salary_range: str | None = None
    company_type: str = "互联网"
    is_active: bool = True


class JobUpdate(BaseModel):
    title: str | None = None
    category: str | None = None
    level: str | None = None
    description: str | None = None
    requirements: dict | None = None
    skill_tree: dict | None = None
    salary_range: str | None = None
    company_type: str | None = None
    is_active: bool | None = None
